sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,502 @@
+  signingPolicy,
+  userAgentPolicy,
+  logPolicy,
+  proxyPolicy,
+  atomSerializationPolicy,
+  RequestPolicyFactory,
+  AtomXmlSerializer,
+  URLBuilder,
+  ProxySettings
+} from "@azure/core-http";
+
+import { httpAtomXml } from "./log";
+import { SasServiceClientCredentials } from "./util/sasServiceClientCredentials";
+import * as Constants from "./util/constants";
+
+import { QueueResourceSerializer, QueueOptions } from "./serializers/queueResourceSerializer";
+import { TopicResourceSerializer, TopicOptions } from "./serializers/topicResourceSerializer";
+import {
+  SubscriptionResourceSerializer,
+  SubscriptionOptions
+} from "./serializers/subscriptionResourceSerializer";
+import { RuleResourceSerializer, RuleOptions } from "./serializers/ruleResourceSerializer";
+
+interface ListRequestOptions {
+  top?: number;
+  skip?: number;
+}
+
+/**
+ * All operations return a `Promise<HttpOperationResponse>` object of which the `parsedBody`
+ * is populated with the response in JSON format with following structure:
+ *  {
+ *   response: <actual response>,
+ *   result: <parsed result, if any>,
+ *   error: <error information, if any>
+ *  }
+ */
+export class ServiceBusAtomManagementClient extends ServiceClient {
+  private endpoint: string;
+
+  private queueResourceSerializer: AtomXmlSerializer;
+  private topicResourceSerializer: AtomXmlSerializer;
+  private subscriptionResourceSerializer: AtomXmlSerializer;
+  private ruleResourceSerializer: AtomXmlSerializer;
+
+  /**
+   * Initializes a new instance of the ServiceBusManagementClient class.
+   * @param credentials Credentials needed for the client to connect to Azure.
+   * @param options The parameter options
+   */
+  constructor(connectionString: string, proxySettings?: ProxySettings) {
+    const connectionStringObj = ServiceBusAtomManagementClient.parseConnectionString(
+      connectionString
+    );
+    const credentials = new SasServiceClientCredentials(
+      connectionStringObj.SharedAccessKeyName,
+      connectionStringObj.SharedAccessKey
+    );
+
+    const requestPolicyFactories: RequestPolicyFactory[] = [];
+    requestPolicyFactories.push(userAgentPolicy());
+    requestPolicyFactories.push(logPolicy(httpAtomXml));
+    requestPolicyFactories.push(atomSerializationPolicy());
+    requestPolicyFactories.push(signingPolicy(credentials));
+
+    if (proxySettings) {
+      requestPolicyFactories.push(proxyPolicy(proxySettings));
+    }
+    const serviceClientOptions: ServiceClientOptions = {
+      requestPolicyFactories: requestPolicyFactories
+    };
+
+    super(credentials, serviceClientOptions);
+    this.queueResourceSerializer = new QueueResourceSerializer();
+    this.topicResourceSerializer = new TopicResourceSerializer();
+    this.subscriptionResourceSerializer = new SubscriptionResourceSerializer();
+    this.ruleResourceSerializer = new RuleResourceSerializer();
+    this.endpoint = (connectionString.match("Endpoint=sb://(.*)/;") || "")[1];
```

Comment:
Since `parseConnectionString` has already done the job of parsing the connection string, how about using the parsed output?
```
if (connectionStringObj.Endpoint && connectionStringObj.Endpoint.startsWith("sb://") {
   this.endpoint = connectionStringObj.substr(5);
}
```

Also, if the Endpoint was not provided in the connection string, are we doing any validation and throwing error up front during client creation or waiting until the http request that is made to fail?

Reply:
> how about using the parsed output?

`substr(5)` is not enough and the endpoint value needs to be parsed anyway

> waiting until the http request that is made to fail?

Yes, doing the same checks as we have been

Reply:
When the connection string is malformed and is missing the Endpoint, we will continue to make the request using the url `https://undefined/queuename` which will then give a 404 most probably.

It is good to exit early when we know that the endpoint is missing.

Please consider checking `connectionStringObj.Endpoint` and throw an error if it is not present